guest/api/clusterevent.go

```diff
@@ -0,0 +1,77 @@
+package api
+
+import "strconv"
+
+// ActionType is an integer to represent one type of resource change.
+// Different ActionTypes can be bit-wised to compose new semantics.
+type ActionType uint32
+
+// Constants for ActionTypes.
+const (
+	Add    ActionType = 1 << iota // 1
+	Delete                        // 10
+	// UpdateNodeXYZ is only applicable for Node events.
+	UpdateNodeAllocatable // 100
+	UpdateNodeLabel       // 1000
+	UpdateNodeTaint       // 10000
+	UpdateNodeCondition   // 100000
+
+	All ActionType = 1<<iota - 1 // 111111
+
+	// Use the general Update type if you don't either know or care the specific sub-Update type to use.
+	Update = UpdateNodeAllocatable | UpdateNodeLabel | UpdateNodeTaint | UpdateNodeCondition
+)
+
+// GVK is short for group/version/kind, which can uniquely represent a particular API resource.
+type GVK uint32
+
+// Constants for GVKs.
+const (
+	Pod GVK = iota
+	Node
+	PersistentVolume
+	PersistentVolumeClaim
+	PodSchedulingContext
+	ResourceClaim
+	StorageClass
+	CSINode
+	CSIDriver
+	CSIStorageCapacity
+	WildCard
+)
+
+// String implements fmt.Stringer
+func (gvk GVK) String() string {
```

Comment:
I can remove this if people don't like it. If we keep it, I will test it. let me know.